searx/plugins/__init__.py

```diff
@@ -63,9 +64,17 @@ def register(self, *plugins, external=False):
             plugins = load_external_plugins(plugins)
         for plugin in plugins:
             for plugin_attr, plugin_attr_type in required_attrs:
-                if not hasattr(plugin, plugin_attr) or not isinstance(getattr(plugin, plugin_attr), plugin_attr_type):
+                if not hasattr(plugin, plugin_attr):
                     logger.critical('missing attribute "{0}", cannot load plugin: {1}'.format(plugin_attr, plugin))
                     exit(3)
+                attr = getattr(plugin, plugin_attr)
+                if not isinstance(attr, plugin_attr_type):
+                    type_attr = str(type(attr))
+                    logger.critical(
+                        'attribute "{0}" is of type {2}, must be of type {3}, cannot load plugin: {1}'
+                        .format(plugin_attr, plugin, type_attr, plugin_attr_type)
```

Comment:
Nit: I think string replacement by log-function arguments [`*args`](https://docs.python.org/3/library/logging.html#logging.Logger.debug) consumes less resources when logging is disabled, while the format operation of the string is always executed, even if logging is disabled.

Reply:
Yes, clearly.
`sys.exit()`  just after "hides / solves" the issue.